### PR TITLE
xinit: update to 1.4.2.

### PR DESCRIPTION
**Pull request by motorto**
#### Testing the changes
- I tested the changes in this PR: **YES**, x86_64

#### Local build testing
- I built this PR locally for my native architecture, x86_64
```
SUMMARY
pkg:xinit host:x86_64 target:x86_64 cross:n result:OK
pkg:xinit host:x86_64 target:aarch64-musl cross:y result:OK
```

### CODE REVIEW

**Review by vincele — APPROVED** [review]
LGTM
serverauth.XXXXXXXXXX is in /tmp

File: srcpkgs/xinit/patches/fs25361.patch [hunk 1]
```diff
@@ -1,6 +1,7 @@
---- a/startx.cpp	2018-03-10 02:46:03.000000000 +0100
-+++ b/startx.cpp	2018-03-12 20:30:43.041201031 +0100
-@@ -296,9 +296,9 @@
+diff -u xinit.orig/startx.cpp xinit/startx.cpp
+--- xinit.orig/startx.cpp	2011-08-02 18:26:57.161711018 +0400
++++ xinit/startx.cpp	2011-08-02 19:06:39.348816866 +0400
```
Comment (kruceter) [inline srcpkgs/xinit/patches/fs25361.patch:3]: Revert changes made to the selected lines.

File: srcpkgs/xinit/template [hunk 2]
```diff
@@ -10,8 +10,8 @@ short_desc="X init program"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="MIT"
 homepage="https://xorgwiki.freedesktop.org/xorg"
```
Comment (classabbyamp) [inline srcpkgs/xinit/template:12]:
```suggestion
homepage="https://gitlab.freedesktop.org/xorg/app/xinit"
```

Comment (motorto) [reply srcpkgs/xinit/template]: Thanks, solved this.

**Pull request merged**